gammapy/background/models.py

```diff
@@ -71,3 +83,699 @@ def evaluate(self, x, y):
         y = np.asanyarray(y, dtype=float)
         parvals = self.parvals(x)
         return self._evaluate_y(y, parvals)
+
+
+def _make_bin_edges_array(lo, hi):
+    """Make bin edges array from a low values and a high values array.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
+
+    Parameters
+    ----------
+    lo : `~numpy.array`
+        lower boundaries
+    hi : `~numpy.array`
+        higher boundaries
+
+    Returns
+    -------
+    bin_edges : `~numpy.array`
+        array of bin edges as [[low], [high]]
+    """
+    return np.append(lo.flatten(), hi.flatten()[-1:])
+
+
+def _parse_bg_units(background_unit):
+    """
+    Utility function to parse the bg units correctly.
+    """
+    # try 1st to parse them as astropy units
+    try:
+        u.Unit(background_unit)
+    # if it fails, try to parse them manually
+    except ValueError:
+        tev_units = ['1/s/TeV/sr', 's-1 sr-1 TeV-1', '1 / (s sr TeV)',
+                     '1 / (TeV s sr)']
+        mev_units = ['1/s/MeV/sr', 'MeV-1 s-1 sr-1', '1 / (s sr MeV)',
+                     '1 / (MeV s sr)']
+        if background_unit in tev_units:
+            background_unit = '1 / (s TeV sr)'
+        elif background_unit in mev_units:
+            background_unit = '1 / (s MeV sr)'
+        # if it still fails, raise an exception
+        else:
+            raise ValueError("Cannot interpret units ({})".format(background_unit))
+
+    return background_unit
+
+
+class CubeBackgroundModel(object):
+
+    """Cube background model.
+
+    Container class for cube background model (X, Y, energy).
+    (X, Y) are detector coordinates (a.k.a. nominal system).
+    The class hass methods for reading a model from a fits file,
+    write a model to a fits file and plot the models.
+
+    The order of the axes in the background cube is `(E, y, x)`,
+    so in order to access the data correctly, the call is
+    `bg_cube_model.background[energy_bin, dety_bin, detx_bin]`.
+
+    Parameters
+    ----------
+    detx_bins : `~astropy.coordinates.Angle`
+        Spatial bin edges vector (low and high). X coordinate.
+    dety_bins : `~astropy.coordinates.Angle`
+        Spatial bin edges vector (low and high). Y coordinate.
+    energy_bins : `~astropy.units.Quantity`
+        Energy bin edges vector (low and high).
+    background : `~astropy.units.Quantity`
+        Background cube in (energy, X, Y) format.
+
+    Examples
+    --------
+    Access cube bg model data:
+
+    .. code:: python
+
+        energy_bin = bg_cube_model.find_energy_bin(energy=Quantity(2., 'TeV'))
+        det_bin = bg_cube_model.find_det_bin(det=Angle([0., 0.], 'degree'))
+        bg_cube_model.background[energy_bin, det_bin[1], det_bin[0]]
+    """
+    # TODO: how can I link to this example in the models.rst doc subpage?!!!
```

Comment:
The only way to achieve this would be a `..literalinclude` of an extra `.py` file from the docstring and the high-level docs. I've never seen anyone use `..literalinclude` from a docstring, so I'm not sure if it works. And anyways it's not a good solution because in interactive use from IPython people wouldn't see the code example.

So you'll have to pick: duplicate the code example or add a sentence linking to it in another location. I think both are OK.